### PR TITLE
Story add

### PR DESCRIPTION
**Pull request by ajhswan**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -40,3 +42,39 @@ function displayData (array $query):string{
...
+function testInput(string $data): string {
+    $data = trim($data);
+    $data = filter_var($data, FILTER_SANITIZE_STRING, FILTER_FLAG_ENCODE_LOW | FILTER_FLAG_ENCODE_AMP);
+    return $data;
+}
+
+
+/**Insert sanitised data from user form into database
+ *
+ * @param string $sName
```
Comment (charliecog) [inline functions.php]: need descriptions for all the params

File: functions.php [hunk 2]
```diff
@@ -40,3 +42,39 @@ function displayData (array $query):string{
...
+ * @param string $details
+ * @param string $amount
+ * @param string $ccy
+ * @param string $date
+ * @param PDO $db
+ *
+ * @return bool
```
Comment (charliecog) [inline functions.php]: need return description

File: functions.php [hunk 3]
```diff
@@ -40,3 +42,39 @@ function displayData (array $query):string{
         return $result;
 }
 
+/** Validate and sanitise $_POST data from user form submission
+ *
+ * @param mixed $data
```
Comment (charliecog) [inline functions.php]: descriptions

File: functions.php [hunk 4]
```diff
@@ -40,3 +42,39 @@ function displayData (array $query):string{
...
+ *
+ * @return mixed
```
Comment (charliecog) [inline functions.php]: description

File: functions.php [hunk 5]
```diff
@@ -40,3 +42,39 @@ function displayData (array $query):string{
...
+ * @param string $details
+ * @param string $amount
+ * @param string $ccy
+ * @param string $date
+ * @param PDO $db
+ *
+ * @return bool
+ */
+function insertData(string $sName, string $details, string $amount, string $ccy, string $date, PDO $db): bool {
+
```
Comment (charliecog) [inline functions.php]: remove line break

File: styles.css [hunk 6]
```diff
@@ -39,45 +40,141 @@ body {
...
+    padding: 0;
+}
+
+input[type=text] {
+    width: 80%;
+}
+
+input[type=submit] {
+     width: 30%;
+     margin: 10px auto;
```
Comment (charliecog) [inline styles.css:141]: need to add cursor: pointer;

File: styles.css [hunk 7]
```diff
@@ -39,45 +40,141 @@ body {
...
+}
+
+.button:hover {
+    background: cornflowerblue;
+}
+
+.overlay {
+    position: fixed;
+    top: 0;
+    bottom: 0;
```
Comment (charliecog) [inline styles.css]: dont need - remove

**Review by charliecog — COMMENTED** [review]
few last bits to do

if you aren't doing error messages it would be worth detailing on the form what is expected so that its clearer if form fails to add to db

File: functions.php (hunk 1 above)
Comment (ajhswan) [reply functions.php]: done

File: functions.php (hunk 2 above)
Comment (ajhswan) [reply functions.php]: done

File: functions.php (hunk 3 above)
Comment (ajhswan) [reply functions.php]: done

File: functions.php (hunk 4 above)
Comment (ajhswan) [reply functions.php]: done

File: functions.php (hunk 5 above)
Comment (ajhswan) [reply functions.php]: done

File: styles.css (hunk 6 above)
Comment (ajhswan) [reply styles.css:141]: done

File: styles.css (hunk 7 above)
Comment (ajhswan) [reply styles.css]: removed

**Pull request merged**